browser/resources/settings/brave_settings_overrides.js

```diff
@@ -1,9 +1,317 @@
+  const menuEl = document.createElement('a')
+  menuEl.href = href
+  menuEl.innerHTML = `
+    <iron-icon icon="${iconName}"></iron-icon>
+    ${title}
+  `
+  return menuEl
+}
+
+function getMenuElement (templateContent, href) {
+  const menuEl = templateContent.querySelector(`a[href="${href}"]`)
+  if (!menuEl) {
+    console.error(`[Brave Settings Overrides] Could not find menu item '${href}'`)
+  }
+  return menuEl
+}
+
+function getSectionElement (templateContent, sectionName) {
+  const sectionEl = templateContent.querySelector(`template[if="[[showPage_(pageVisibility.${sectionName})]]"]`)
+  if (!sectionEl) {
+    console.error(`[Brave Settings Overrides] Could not find section '${sectionName}'`)
+  }
+  return sectionEl
+}
+
+if (!BravePatching) {
+  console.error('BravePatching was not available to brave_settings_overrides.js')
+}
+
+//
+// Override, extend or modify existing modules
+//
+
+const BraveClearSettingsMenuHighlightBehavior = {
+  ready: function() {
+    // Clear menu selection after scrolling away.
+    // Chromium's menu is not persistant, so does not have
+    // this issue.
+    const container = this.$.container
+    if (!container) {
+      console.error('Could not find #container in settings-ui module')
+    }
+    const menu = this.$$('settings-menu')
+    if (!menu) {
+      console.error('Could not find settings-menu in settings-ui module')
+    }
+    let onScroll
+    function stopObservingScroll() {
+      if (onScroll) {
+        container.removeEventListener('scroll', onScroll)
+        onScroll = null
+      }
+    }
+    window.addEventListener('showing-section', ({ detail: section }) => {
+      // Currently showing or about to scroll to `section`.
+      // If we're getting further away from section top
+      // then section is no longer 'selected'.
+      // TODO(petemill): If this wasn't a chromium module, we'd simply add a handler
+      // for scrolling away, or have the menu change selection as we scroll.
+      stopObservingScroll()
+      function calcDistance() {
+        const sectionScrollTop = section.offsetTop
+        const currentScrollTop = container.scrollTop
+        return Math.abs(sectionScrollTop - currentScrollTop)
+      }
+      let distance = calcDistance()
+      onScroll = throttle(() => {
+        const latestDistance = calcDistance()
+        if (latestDistance > distance) {
+          menu.setSelectedUrl_('')
+          stopObservingScroll()
+        } else {
+          distance = latestDistance
+        }
+      }, 100)
+      container.addEventListener('scroll', onScroll)
+    })
+  }
+}
+
+// Polymer Component Behavior injection (like superclasses)
+BravePatching.RegisterPolymerComponentBehaviors({
   'settings-clear-browsing-data-dialog': [
     BraveClearBrowsingDataOnExitBehavior
+  ],
+  'settings-ui': [
+    BraveClearSettingsMenuHighlightBehavior
   ]
 })
+
+// Templates
+BravePatching.RegisterPolymerTemplateModifications({
+  'settings-ui': (templateContent) => {
```

Comment:
for other reviewers: the keys here (ex: `settings-ui`) are the names of components on the page 😄